kof-operator/internal/controller/configmap_controller_test.go

```diff
@@ -0,0 +1,382 @@
+					Namespace: ReleaseNamespace,
+					Labels: map[string]string{
+						KofRecordRulesClusterNameLabel: "regional1",
+					},
+				},
+				Data: map[string]string{
+					"record-group1": `- expr: count (up{cluster="child2"} == 1)
+  record: count:child2_up1
+- expr: count (up{cluster="child3"} == 1)
+  record: count:child3_up1`,
+				},
+			}
+			Expect(k8sClient.Create(ctx, clusterRecordConfigMap)).To(Succeed())
+
+			By("creating record VMules ConfigMap")
+			recordVMRulesConfigMap := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      recordVMRulesConfigMapName,
+					Namespace: ReleaseNamespace,
+					Labels: map[string]string{
+						KofRecordVMRulesClusterNameLabel: "regional1",
+						utils.KofGeneratedLabel:          "true",
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, recordVMRulesConfigMap)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			configMap := &corev1.ConfigMap{}
+			prometheusRule := &promv1.PrometheusRule{}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      prometheusRuleName,
+				Namespace: ReleaseNamespace,
+			}, prometheusRule); err == nil {
+				By("deleting PrometheusRule")
+				Expect(k8sClient.Delete(ctx, prometheusRule)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      defaultAlertConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting default alert ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      clusterAlertConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting cluster alert ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      promxyRulesConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting promxy rules ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      defaultRecordConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting default record ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      clusterRecordConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting cluster record ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+
+			if err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      recordVMRulesConfigMapName,
+				Namespace: ReleaseNamespace,
+			}, configMap); err == nil {
+				By("deleting record VMules ConfigMap")
+				Expect(k8sClient.Delete(ctx, configMap)).To(Succeed())
+			}
+		})
+
+		It("should successfully reconcile ConfigMaps", func() {
```

Comment:
Please add few more tests: 

- when the configmap is deleted
- when the configmap is invalid